File: packages/health_connector_hc_android/android/src/main/kotlin/com/phamtunglam/health_connector_hc_android/mappers/health_measurement_unit_mappers/PercentageMapper.kt
package com.phamtunglam.health_connector_hc_android.mappers.health_measurement_unit_mappers

import androidx.health.connect.client.units.Percentage
import com.phamtunglam.health_connector_hc_android.pigeon.PercentageDto

/**
 * Converts a [PercentageDto] to a Health Connect [Percentage] object.
 */
internal fun PercentageDto.toHealthConnect(): Percentage = Percentage(decimal)

/**
 * Converts a Health Connect [Percentage] object to a [PercentageDto].
 *
 * Uses decimal as the transfer unit for consistency (0.0 to 1.0).
 */
internal fun Percentage.toDto(): PercentageDto = PercentageDto(
    decimal = value,
)

File: packages/health_connector_hc_android/lib/src/mappers/measurement_unit_mappers/percentage_mapper.dart
import 'package:health_connector_core/health_connector_core_internal.dart'
    show Percentage;
import 'package:health_connector_hc_android/src/pigeon/health_connector_hc_android_api.g.dart'
    show PercentageDto;
import 'package:meta/meta.dart' show internal;

/// Converts [Percentage] to [PercentageDto].
@internal
extension PercentageToDto on Percentage {
  PercentageDto toDto() {
    // Uses decimal as the transfer unit for consistency.
    return PercentageDto(
      decimal: asDecimal,
    );
  }
}

/// Converts [PercentageDto] to [Percentage].
@internal
extension PercentageDtoToDomain on PercentageDto {
  Percentage toDomain() {
    return Percentage.fromDecimal(decimal);
  }
}
